go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -1453,6 +1454,104 @@ func (s *VtctldServer) GetSchema(ctx context.Context, req *vtctldatapb.GetSchema
 	}, nil
 }
 
+func (s *VtctldServer) GetSchemaMigrations(ctx context.Context, req *vtctldatapb.GetSchemaMigrationsRequest) (resp *vtctldatapb.GetSchemaMigrationsResponse, err error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.GetShard")
+	defer span.Finish()
+
+	defer panicHandler(&err)
+
+	span.Annotate("keyspace", req.Keyspace)
+
+	var condition string
+	switch {
+	case req.Uuid != "":
+		span.Annotate("uuid", req.Uuid)
+		if !schema.IsOnlineDDLUUID(req.Uuid) {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "%s is not a valid UUID", req.Uuid)
+		}
+
+		condition, err = sqlparser.ParseAndBind("migration_uuid=%a", sqltypes.StringBindVariable(req.Uuid))
+	case req.MigrationContext != "":
+		span.Annotate("migration_context", req.MigrationContext)
+		condition, err = sqlparser.ParseAndBind("migration_context=%a", sqltypes.StringBindVariable(req.MigrationContext))
+	case req.Status != vtctldatapb.SchemaMigration_UNKNOWN:
+		span.Annotate("migration_status", schematools.SchemaMigrationStatusName(req.Status))
+		condition, err = sqlparser.ParseAndBind("migration_status=%a", sqltypes.StringBindVariable(schematools.SchemaMigrationStatusName(req.Status)))
+	case req.Recent != nil:
+		var d time.Duration
+		d, _, err = protoutil.DurationFromProto(req.Recent)
+		if err != nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "error parsing duration: %s", err)
+		}
+
+		span.Annotate("recent", d.String())
+		condition = fmt.Sprintf("requested_timestamp > now() - interval %0.f second", d.Seconds())
+	default:
+		condition = "migration_uuid like '%'"
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Error generating OnlineDDL query: %+v", err)
+	}
+
+	order := " order by `id` "
+	switch req.Order {
+	case vtctldatapb.QueryOrdering_DESCENDING:
+		order += "DESC"
+	default:
+		order += "ASC"
+	}
+
+	var skipLimit string
+	if req.Limit > 0 {
+		skipLimit = fmt.Sprintf("LIMIT %v,%v", req.Skip, req.Limit)
+		span.Annotate("skip_limit", skipLimit)
+	}
+
+	query := selectSchemaMigrationsQuery(condition, order, skipLimit)
+
+	tabletsResp, err := s.GetTablets(ctx, &vtctldatapb.GetTabletsRequest{
+		Cells:      nil,
+		Strict:     false,
+		Keyspace:   req.Keyspace,
+		TabletType: topodatapb.TabletType_PRIMARY,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	results := map[string]*sqltypes.Result{}
+	for _, tablet := range tabletsResp.Tablets {
+		alias := topoproto.TabletAliasString(tablet.Alias)
+
+		// TODO: execute in parallel
```

Comment:
I feel like this is worth doing now as you can have 100s of shards with each query taking 100s of ms (and one could take seconds e.g.).